.gitignore

```diff
@@ -10,14 +10,15 @@
 ._.DS_Store
 .htpasswd
 .pidfile
-npm-debug.log
 .digest-cache.json
 check.txt
 .vagrant
 *.heapsnapshot
 
-# YARN local install
+# package managers
 .yarn
+npm-debug.log
+package-lock.json
```

Comment:
For the record: https://github.com/npm/npm/blob/v5.0.0/doc/files/package-lock.json.md

What I take from this is:

- The top level project will not use it.
- `npm publish` will ignore it.
- But for MAGE development, it can be considered useful.
- So yes, commit it.